scripts/zones/Eastern_Adoulin/npcs/Octavien.lua

```diff
@@ -0,0 +1,56 @@
+-----------------------------------
+-- Area: Eastern Adoulin (257)
+--  NPC: Octavien
+-- Type: Palace Guard
+-- Starts Children of the Rune
+-- !pos 100.580 -40.150 -63.830
+-----------------------------------
+require("scripts/globals/keyitems")
+require("scripts/globals/quests")
+local ID = require("scripts/zones/Eastern_Adoulin/IDs")
+
+function onTrade(player,npc,trade)
+end
+
+function onTrigger(player,npc)
+  -- CHILDREN OF THE RUNE
+  local childrenOfTheRuneQuestStatus = player:getQuestStatus(ADOULIN, tpz.quest.id.adoulin.CHILDREN_OF_THE_RUNE)
+  if ENABLE_SOA == 1 and childrenOfTheRuneQuestStatus == QUEST_AVAILABLE and player:getMainLvl() >= ADVANCED_JOB_LEVEL then
```

Comment:
If Octavien is flagged properly in the database, he won't even spawn if SoA is off.

You already would need to get into SoA and the NPCs involved with that shouldn't work with that off, so checking ENABLE_SOA isn't needed here.

Reply:
> Granted you probably couldn't get to Adoulin if the expansion is flag is off,

The starting mission that would let you arrive here should not activate,.

> It seems like these flags in Lua were intended to give runtime control for enabling/disabling expansions:

Generally those settings are to prevent missions activating or prevent working NPCs confusing the players. They really shouldn't ever be disabled after the fact. I was not aware anyone had started adding them into job quests..

the whole point of my comment:
`INSERT INTO npc_list VALUES (17830003,'Octavien','Octavien',128,2.444,-11.442,-16.595,20,40,40,0,0,0,0,27,0x01000F0352115221523152415251006000700000,32,'SOA',1);`
If you are **actually** using the enable flags to turn off content, _the npc then doesn't even spawn_ because its flagged 'SOA' there. :man_shrugging: 

> -- Setting to lock content more accurately to the content you have defined above
-- This generally results in a more accurate presentation of your selected expansions
-- as well as a less confusing player experience for things that are disabled (things that are disabled are not loaded)
-- This feature correlates to the content_tag column in the SQL files
RESTRICT_CONTENT = 0

I mean, is anyone actually going to disable SOA and expect to do other things but then not allow to trigger the job quest? I would expect that if I am allowed into the zone, and the NPCs are up, its active. 

I guess we need to ask the devs what they think about including expansion checks on job quests. Feels redundant to me but either way we'll want them all consistent (all exp jobs do it or none do it, excluding zilart of course - no flag for it).

Reply:
I wonder if those Lua settings flags are even needed at all then. They don't seem to be populated from the SQL, as far as I can tell.

But for the scope of this change, I'm happy to do whatever the consensus is. I'm also a big fan of consistency, so if folks prefer to keep these flag checks then we can leave it. Otherwise if the preference is to remove them then I'm happy to do so.

I actually prefer to remove the flag, if it's truly not needed, as it simplifies the code.

Reply:
> I wonder if those Lua settings flags are even needed at all then. They don't seem to be populated from the SQL, as far as I can tell.

what? The core looks ate the lua setting, and then find a matching sql tag to decide if the NPC spawns or not. That can't work without both parts involved. We even piggy backed that for the FoV books - by making a setting named `ENABLE_thing` and their expansion tag in the NPC list  `thing` you use it for any grouping of NPCs with the same matching tag.

> But for the scope of this change, I'm happy to do whatever the consensus is. 

I asked in discord last night but they didn't comment here. 

@ibm2431 I summon thee.

Reply:
> The core looks at the lua setting, and then find a matching sql tag to decide if the NPC spawns or not.

Would you mind sharing some links for my own learning? I couldn't find that just by searching around for the Lua flag name.

Reply:
https://github.com/project-topaz/topaz/blob/8985d2af68876367592222836db1f8e5b15c8175/src/map/lua/luautils.cpp#L1201

```cpp
            std::string contentVariable("ENABLE_");
            contentVariable.append(contentTag);
```

Reply:
and here's where it hits entity loading from sql

https://github.com/project-topaz/topaz/blob/5991d15a9a17549ca7a076b7a4a42ee3a26be262/src/map/utils/zoneutils.cpp#L279

```cpp
    if (ret != SQL_ERROR && Sql_NumRows(SqlHandle) != 0)
    {
        while(Sql_NextRow(SqlHandle) == SQL_SUCCESS)
        {
            const char* contentTag = (const char*)Sql_GetData(SqlHandle, 16);

            if (luautils::IsContentEnabled(contentTag) == false)
            {
                continue;
            }
```

Reply:
Sounds good. Removed!